core/domain/question_domain.py

```diff
@@ -220,14 +250,22 @@ def update_state_from_model(
                     state data.
             current_state_schema_version: int. The current state
                 schema version.
+            question_model: QuestionModel or None. The question model loaded
+                from the datastore. None indicates it is not needed in this
+                function.
         """
         versioned_question_state['state_schema_version'] = (
             current_state_schema_version + 1)
 
         conversion_fn = getattr(cls, '_convert_state_v%s_dict_to_v%s_dict' % (
             current_state_schema_version, current_state_schema_version + 1))
-        versioned_question_state['state'] = conversion_fn(
-            versioned_question_state['state'])
+        if question_model and current_state_schema_version == 29:
+            versioned_question_state['state'] = conversion_fn(
+                versioned_question_state['state'],
+                question_model.linked_skill_ids)
```

Comment:
Instead of doing this we should make a method which will allow us to convert the entire question object, like the way we do in the exploration domain. I'm not sure, but can you please check?

Reply:
I think this is how it is being done currently in the codebase. I will explore this part later and if it's better I will change it in a separate PR.